Constructor changes

isopropylamine opened the pull request.

Jerminater committed:
Merge pull request #2 from isopropylamine/master

Merge pull request #1 from isopropylamine/Jeremy's-suggested-changes

isopropylamine committed:
Changed constructor and compareTo logic

I used setScale to round and check if the values are equal to bigDecimal.ZERO, and placed it in the constructor

I'm still getting issues where it's missing a space for Indian currency, and I don't know what's wrong.

Jerminater commented on src/CurrencyViewModel.java:
```diff
@@ -18,6 +19,14 @@
     public CurrencyViewModel(double amount, Locale locale) {
         this.amount = new BigDecimal(amount);
```
setting the scale here would eliminate the use of a new variable `roundedAmount`

Jerminater commented on src/CurrencyViewModel.java:
```diff
@@ -18,6 +19,14 @@
         this.numberFormat = NumberFormat.getCurrencyInstance(locale);
+
+        //The rounded amount will change the currency amount to one with 2 digits after the decimal
+        BigDecimal roundedAmount = this.amount.setScale(2, RoundingMode.HALF_UP);
```
use the roundingMode, and the maximumFractionDigits of number format for the scale.

Jerminater commented on .idea/libraries/junit_platform_console_standalone_1_6_2.xml:
```diff
@@ -1,10 +1,15 @@
 <component name="libraryTable">
   <library name="junit-platform-console-standalone-1.6.2">
     <CLASSES>
```
This file should be in your gitignore

Jerminater replied:
The whole .idea folder should be in git ignore actually. it is unique to your project alone

isopropylamine replied:
Thanks! I didn't know about gitignore!

Jerminater commented on .idea/misc.xml:
```diff
@@ -3,7 +3,7 @@
   <component name="ProjectKey">
     <option name="state" value="project://e2804f05-5315-4fc6-a121-c522a6c26470" />
```
this file should be in git ignore.

isopropylamine committed:
Update CurrencyViewModel.java

Removed the extra variable, and changed the scale to the maximum digits of the locale's currency.

isopropylamine committed:
Revert "Update CurrencyViewModel.java"

This reverts commit 17ce349c773a69432a0c59bc292ee53da845be95.

isopropylamine committed:
Update CurrencyViewModel.java

Removed extra variable, and changed the scale of the amount to that of the given locale's currency maximum fraction digits.

isopropylamine committed:
Update CurrencyViewModel.java

isopropylamine committed:
Update CurrencyViewModel.java

isopropylamine committed:
Update CurrencyViewModel.java

isopropylamine committed:
Merge branch 'master' into Constructor-Changes

isopropylamine committed:
Update CurrencyViewModel.java
